packages/react-composites/src/composites/MeetingComposite/ChatButtonLogicWrapper.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ChatMessage } from '@azure/communication-chat';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { ChatAdapter } from '../ChatComposite';
+import { ChatButton, ChatButtonProps } from './ChatButton';
+
+/**
+ * @private
+ */
+export type ChatButtonLogicWrapperProps = {
+  chatButtonProps: ChatButtonProps;
+  chatAdapter: ChatAdapter;
+  showChat: boolean;
+  onChatButtonClicked: () => void;
+  disabled: boolean;
+};
+
+/**
+ * @private
+ */
+export const ChatButtonLogicWrapper = (props: ChatButtonLogicWrapperProps): JSX.Element => {
+  const { chatButtonProps, chatAdapter, showChat, onChatButtonClicked, disabled } = props;
+  const [unreadChatMessagesCount, setUnreadChatMessagesCount] = useState<number>(0);
+  useEffect(() => {
+    if (showChat === true) {
+      setUnreadChatMessagesCount(0);
+      return;
+    }
+    const incrementUnreadtChatMessagesCount = (event: { message: ChatMessage }): void => {
+      if (!showChat && event.message.senderDisplayName !== '') {
+        if (event.message.type === 'text' || event.message.type === 'html') {
+          setUnreadChatMessagesCount(unreadChatMessagesCount + 1);
+        }
+      }
+    };
+    chatAdapter.on('messageReceived', incrementUnreadtChatMessagesCount);
+
+    return () => {
+      chatAdapter.off('messageReceived', incrementUnreadtChatMessagesCount);
+    };
+  }, [chatAdapter, setUnreadChatMessagesCount, showChat, unreadChatMessagesCount]);
+
+  return (
+    <ChatButton
```

Comment:
I almost expected this to look like: `<ChatButton {...chatButtonProps} unreadMessageCount={unreadMessageCount} />`

Any reason why all the props needed by `ChatButton` aren't provided inside `chatButtonProps`. The only `prop`s that this component uses (and `ChatButton` doesn't use) are `chatAdapter` and `showChat`, so the props for this component should be `{chatAdapter, isChatPaneVisible, chatButtonProps}`, and this component should never look inside `chatButtonProps`.

Reply:
the items that aren't in the chatButtonProps object aren't actually in the type ChatButtonProps type so it was just erroring out unless I passed them in like this. open to suggestions to how to do this with the spread though!